node/config/migrate.go

```diff
@@ -0,0 +1,177 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	logging "github.com/ipfs/go-log/v2"
+)
+
+var log = logging.Logger("cfg")
+
+// CurrentVersion is the config version expected by Boost.
+// We need to migrate the config file to this version.
+const CurrentVersion = 1
+
+type migrateUpFn = func(cfgPath string) (string, error)
+
+var migrations = []migrateUpFn{
+	v0Tov1,
+}
+
+// Migrate from config version 0 to version 1
+func v0Tov1(cfgPath string) (string, error) {
+	cfg, err := FromFile(cfgPath, DefaultBoost())
+	if err != nil {
+		return "", fmt.Errorf("parsing config file %s: %w", cfgPath, err)
+	}
+
+	boostCfg, ok := cfg.(*Boost)
+	if !ok {
+		return "", fmt.Errorf("unexpected config type %T: expected *config.Boost", cfg)
+	}
+
+	// Update the Boost config version
+	boostCfg.ConfigVersion = 1
+
+	// For the migration from v0 to v1 just add the config version and add
+	// comments to the file
+	bz, err := ConfigUpdate(boostCfg, DefaultBoost(), true)
+	if err != nil {
+		return "", fmt.Errorf("applying configuration: %w", err)
+	}
+
+	return string(bz), nil
+}
+
+// This struct is used to get the config file version
+type VersionCfg struct {
+	ConfigVersion int
+}
+
+func ConfigMigrate(repoPath string) error {
```

Comment:
Nitpick: we are in package `config`, we could call this `Migrate`, so that we don't end up with config.ConfigMigrate(...), but just config.Migrate(...)

Reply:
Same for `ConfigUpdate`.

Reply:
I thought about doing the same thing - but ConfigUpdate etc are copied over from the lotus repo, so I thought it would make it easier to port over any future bug fixes there if we keep the names the same